queue/bytes_queue_test.go

```diff
@@ -226,6 +255,30 @@ func TestGetEntryFromInvalidIndex(t *testing.T) {
 	assert.EqualError(t, err, "Index must be grater than zero. Invalid index.")
 }
 
+func TestGetEntryFromIndexOutOfRange(t *testing.T) {
+	t.Parallel()
+
+	// given
+	queue := NewBytesQueue(0, true)
+
+	// when
+	assert.Panics(t, func() { queue.Get(42) })
```

Comment:
sure, we can add some args validation in the constructor. Also args in newBigCache constructor also can be validated, for example MaxEntriesInWindow or MaxEntrySize need to be greater than 0.